src/main/java/pl/mateusz/example/springsecurityproject/user/UserController.java

```diff
@@ -0,0 +1,91 @@
+        model.addAttribute("user", userEditDto);
+        return "edit-form";
+    }
+
+    @PostMapping("/user-profile/{username}/edit")
+    public String editAccount(@PathVariable String username, UserEditDto userEditDto, Model model) {
+        try {
+            userService.editAccount(username, userEditDto);
+        } catch (UserWithEmptyFirstNameException | UserWithEmptyLastNameException e) {
+            logger.info(e.getMessage());
+            model.addAttribute("error", e.getMessage());
+            UserEditDto user = userService.getUserEditDtoByUsername(username).orElseThrow(() -> new UserNotFoundException("Brak użytkownika"));
+            model.addAttribute("user", user);
+            return "edit-form";
+        }
+        return "redirect:/edit-success";
+    }
+
+    @GetMapping("/edit-success")
+    public String editSuccessPage() {
+        return "edit-success";
+    }
+
+    @GetMapping("/promote/{username}")
+    public String promotionsForm(@PathVariable String username, Model model) {
+        UserDisplayDto userToPromote = userService.getUserByUsername(username).orElseThrow(() -> new UserNotFoundException("Nie znaleziono użytkownika"));
+        model.addAttribute("user", userToPromote);
+        return "promote";
+    }
+
+    @GetMapping("/dismiss/{username}")
+    public String dismissForm(@PathVariable String username, Model model) {
+        UserDisplayDto userToDismiss = userService.getUserByUsername(username).orElseThrow(() -> new UserNotFoundException("Nie znaleziono użytkownika"));
+        model.addAttribute("user", userToDismiss);
+        return "dismiss";
+    }
+
+    @PostMapping("/promote/{username}")
+    public String promoteUser(@PathVariable String username, UserDisplayDto userDisplayDto, Model model,
+                              @RequestParam(value = "adminRole", required = false) String adminRole) {
+        if ("true".equals(adminRole)) {
+            userService.promoteUser(username, userDisplayDto);
+        } else {
+            UserDisplayDto userToPromote = userService.getUserByUsername(username)
+                    .orElseThrow(() -> new UserNotFoundException("Nie znaleziono użytkownika"));
+            model.addAttribute("user", userToPromote);
+        }
+        return "redirect:/admin";
+    }
+
+    @PostMapping("/dismiss/{username}")
+    public String dismissUser(@PathVariable String username, UserDisplayDto userDisplayDto, Model model,
+                              @RequestParam(value = "adminRole", required = false) String adminRole) {
+        if ("true".equals(adminRole)) {
+            userService.dismissUser(username, userDisplayDto);
+        } else {
+            UserDisplayDto userToPromote = userService.getUserByUsername(username)
+                    .orElseThrow(() -> new UserNotFoundException("Nie znaleziono użytkownika"));
+            model.addAttribute("user", userToPromote);
+        }
+        return "redirect:/admin";
```

Comment:
Jeżeli niżej robisz redirect, to t rzeczy zapisywane w modelu i tak zostaną utracone